src/Avatar.svelte

```diff
@@ -0,0 +1,50 @@
+<script lang="typescript">
+  import { onMount } from 'svelte';
+  import { createIcon } from '@app/blockies';
+  import { isAddress } from './utils';
+  
+  export let source: string;
+  export let icon: boolean = true;
+  export let glowOnHover = false;
+
+  let container: HTMLElement;
+
+  onMount(() => {
+    if(isAddress(source)) {
+      const seed = source.toLowerCase();
+      const avatar = createIcon({
+        seed,
+        size: 8,
+        scale: 16,
+      });
+      container.style.backgroundImage = `url(${avatar.toDataURL()})`;
+    }
+  });
+</script>
+
+<style>
+  .avatar {
+    border-radius: 50%;
+    min-width: 1rem;
+    min-height: 1rem;
+    width: 100%;
+    height: 100%;
+    background-size: cover;
+    background-repeat: no-repeat;
+  }
+  .avatar.glowOnHover:hover {
+    box-shadow: 0 0 3rem var(--color-secondary);
+  }
+  .icon {
+    display: inline-block;
+    width: 1rem;
+    height: 1rem;
+    margin-right: 0.5rem;
+  }
+</style>
+
+{#if isAddress(source)}
+  <div class="avatar" class:icon bind:this={container} class:glowOnHover title={source}/>
+{:else}
+  <img class="avatar" class:icon src={source} class:glowOnHover alt="avatar"/>
+{/if}
```

Comment:
You should get your editor to add new-lines at the end of files, I'm not sure why it doesn't do that by default (almost all editors do). Files without a newline at the end cause problems with unix tooling, as the last line is not properly terminated.

Reply:
I didn't want my editor to enforce styling since it didn't comply with the format of the code base.
With ESLint I'm sure this will be a thing of the past.

Reply:
I've added this lint to eslint, but it's not really a styling thing - it causes problems when combining files for eg.

Reply:
It seems that eslint is not checking this in .svelte files.. I'll check all those files manually